src/interrupts.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018-2019 CrowdStrike, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+use std::error::{self, Error as InterruptsError};
+use std::fmt::{self, Display};
+use std::io::Cursor;
+use std::mem;
+use std::result;
+
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+
+use common::WHPError;
+use i386::{LapicStateRaw, APIC_REG_OFFSET};
+use platform::VirtualProcessor;
+
+#[derive(Debug)]
+pub enum Error {
+    GetLapic(WHPError),
+    SetLapic(WHPError),
+}
+pub type Result<T> = result::Result<T, Error>;
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        match self {
+            &Error::GetLapic(_) => "GetLapic ioctl failed",
+            &Error::SetLapic(_) => "SetLapic ioctl failed",
+        }
+    }
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Interrupt Error: {}", Error::description(self))
+    }
+}
+
+pub fn get_lapic_reg(lapic: &LapicStateRaw, reg_offset: APIC_REG_OFFSET) -> u32 {
```

Comment:
Why not make all of these member functions?

Reply:
Same as above, was looking to match crosvm/firecracker in their handling of this; some portions come from there and they use functions rather than object methods, and the rest were written to match. Lapic is another good candidate for an abstraction crate, so I suspect both reference implementations will change to match that but at this point I'll keep them written similarly.

Reply:
ok.